actors/builtin/verifreg/verified_registry_actor.go

```diff
@@ -52,17 +52,20 @@ type AddVerifierParams struct {
 }
 
 func (a Actor) AddVerifier(rt vmr.Runtime, params *AddVerifierParams) *adt.EmptyValue {
-	if params.Allowance.LessThan(MinVerifiedDealSize) {
-		rt.Abortf(exitcode.ErrIllegalArgument, "Allowance %d below MinVerifiedDealSize for add verifier %v", params.Allowance, params.Address)
-	}
-
 	var st State
 	rt.State().Readonly(&st)
 	rt.ValidateImmediateCallerIs(st.RootKey)
 
+	verifierIdAddr, err := builtin.ResolveToIDAddr(rt, params.Address)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to resolve verifier address %v to ID address", params.Address)
+
+	if params.Allowance.LessThan(MinVerifiedDealSize) {
+		rt.Abortf(exitcode.ErrIllegalArgument, "Allowance %d below MinVerifiedDealSize for add verifier %v", params.Allowance, verifierIdAddr)
+	}
```

Comment:
Leave the checks that don't require state above the state load call.

Reply:
Done.